Sources/TestingMacros/Support/DiagnosticMessage+Diagnosing.swift

```diff
@@ -36,58 +36,171 @@ func diagnoseIssuesWithTags(in traitExprs: [ExprSyntax], addedTo attribute: Attr
+  // If the first argument to the .bug() trait is unlabelled and a string
+  // literal, check that it can be parsed as a URL or at least as an integer.
+  guard let arg = traitExpr.arguments.first.map(Argument.init),
+        arg.label == nil,
+        let stringLiteralExpr = arg.expression.as(StringLiteralExprSyntax.self),
+        let urlString = stringLiteralExpr.representedLiteralValue else {
+    return
+  }
+
+  if UInt64(urlString) != nil {
+    // The entire URL string can be parsed as an integer, so allow it. Although
+    // the testing library prefers valid URLs here, some bug-tracking systems
+    // might not provide URLs, or might provide excessively long URLs, so we
+    // allow numeric identifiers as a fallback.
+    return
+  }
+
+  func isURLStringValid(_ urlString: String) -> Bool {
+    guard urlString.allSatisfy(\.isASCII),
+          let colonIndex = urlString.firstIndex(of: ":") else {
+      // This can't be a valid URL as far as we're concerned. Exit early without
+      // properly parsing it.
+      return false
+    }
+
+#if SWT_TARGET_OS_APPLE || os(Linux)
+#if !SWT_NO_CURL
+    let url = curl_url()
+    defer {
+      curl_url_cleanup(url)
+    }
+
+    // Attempt to parse the URL.
+    let flags = CUnsignedInt(CURLU_NON_SUPPORT_SCHEME | CURLU_NO_AUTHORITY)
+    switch curl_url_set(url, CURLUPART_URL, urlString, flags) {
+    case CURLUE_OK:
       break
+    case CURLUE_BAD_SLASHES, CURLUE_BAD_SCHEME:
+      // curl does not try to parse URLs without slashes after the colon (see
+      // https://github.com/curl/curl/issues/12205). Work around that constraint
+      // by inserting slashes after the first colon character, on the assumption
+      // we are dealing with a URL like mailto:a@example.com.
+      var urlString = urlString
+      urlString.insert(contentsOf: "//", at: urlString.index(after: colonIndex))
+      return isURLStringValid(urlString)
+    default:
+      // The URL could not be parsed for some other reason.
+      return false
     }
+
+    // Extract the scheme and check that it's not empty.
+    var scheme: UnsafeMutablePointer<CChar>?
+    guard CURLUE_OK == curl_url_get(url, CURLUPART_SCHEME, &scheme, flags), let scheme else {
+      // libcurl won't parse a URL without a scheme given the flags we pass, so
+      // this branch is dead code, but it's not worth asserting over.
+      return false
+    }
+    defer {
+      curl_free(scheme)
+    }
+    return scheme[0] != 0
+#else
+    // libcurl has been disabled.
+    return true
+#endif
+
+#elseif os(WASI)
+    // TODO: URL validation on WASI (this code runs on the host though)
+    return true
+#elseif os(Windows)
+    return urlString.withCString(encodedAs: UTF16.self) { urlString in
+      var components = URL_COMPONENTSW()
+      components.dwStructSize = DWORD(MemoryLayout.size(ofValue: components))
+      return withUnsafeTemporaryAllocation(of: wchar_t.self, capacity: 1) { buffer in
```

Comment:
I don't understand this bit - the capacity is 1, so this allocates 1 `wchar_t` right? not a `wchar_t *`?

Reply:
Yep. I need a `wchar_t` buffer, but I also only need to check if the scheme is empty, so I don't need more than one character.

Reply:
Might be nice to have a comment to that effect.